plot-gui-lib/plot/private/gui/snip2d.rkt

```diff
@@ -33,25 +48,41 @@
     (super-make-object init-bm saved-plot-parameters)
     
     (define (set-message-center)
-      (match-define (vector x-mid y-mid) (rect-center area-bounds-rect))
+      (match-define (vector x-mid y-mid) (rect-center (send area get-area-bounds-rect)))
       (send this set-message-center x-mid y-mid))
     
     (set-message-center)
     
     (define/override (copy)
-      (make-object this%
-        (get-bitmap) (get-saved-plot-parameters)
-        make-bm plot-bounds-rect area-bounds-rect area-bounds->plot-bounds width height))
-    
+      (define c
+        (make-object this%
+                     (get-bitmap) (get-saved-plot-parameters)
+                     make-bm plot-bounds-rect area width height))
+      (when mouse-event-callback
+        (send c set-mouse-event-callback mouse-event-callback))
+      c)
+
     (define left-click-x 0)
     (define left-click-y 0)
     (define left-drag-x 0)
     (define left-drag-y 0)
-    
+
     (define plot-bounds-rects empty)
+
+    (define (area-bounds->plot-bounds rect)
```

Comment:
Looks like this was moved from plot2d.rkt, but the `assert`ions were removed? Was this intended?

Reply:
The `assert` is a typed racket thing, this is non typed code.  I don't know what the equivalent non-typed variant is and I'm not sure it would be valuable in this case.

Reply:
The asserts were checking that `area` and the bounds in `rect` were not false. A non-typed version could be `(unless area (error ....))`.

Removing these asserts looks safe to me, and the function seems to error anyway when I call it with a bad rectangle.